Docker.DotNet/EnumerableQueryStringConverter.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Newtonsoft.Json;
+
+namespace Docker.DotNet
+{
+    /// <summary>
+    /// Handles serialization of objects like Lists, Arrays, etc.
+    /// </summary>
+    internal class EnumerableQueryStringConverter : IQueryStringConverter
+    {
+        public bool CanConvert(Type t)
+        {
+            return typeof (IEnumerable).IsAssignableFrom(t);
+        }
+
+        public string[] Convert(object o)
+        {
+            Debug.Assert(o != null);
+            Debug.Assert(o is IEnumerable);
+
+            var items = new List<string>();
+            foreach (var e in ((IEnumerable) o))
```

Comment:
Unfortunately LINQ only works on IEnumerable<T> and I dont see why we need a generic restriction on EnumerableQueryStringConverter but If you want I can make it EnumerableQueryStringConverter<T> and then on the attributes generate typeof(EnumerableQueryStringConverter<T>) that matches.